### PR TITLE
Upgrade to v1beta3.

### PR DESCRIPTION
**Pull request by pcostell**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,29 +38,29 @@ npm install googleapis
...
+- Maven/Java:
 
 ```
 <dependency>
-  <groupId>com.google.apis</groupId>
-  <artifactId>google-api-services-datastore-protobuf</artifactId>
-  <version>v1beta2-rev1-4.0.0</version>
+  <groupId>com.google.cloud.datastore</groupId>
+  <artifactId>datastore-v1beta3-proto-client</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
```
Comment (pcostell) [inline README.md]: 1.0.0-beta

File: README.md [hunk 2]
```diff
@@ -94,16 +90,14 @@ pip install googledatastore
...
 
 [1]: python/demos/trivial/adams.py
-[2]: java/demos/src/main/java/com/google/api/services/datastore/demos/trivial/Adams.java
-[3]: https://github.com/GoogleCloudPlatform/google-cloud-datastore/blob/master/nodejs/demos/trivial/adams.js
-[4]: https://developers.google.com/datastore
-[5]: https://developers.google.com/datastore/docs/apis/v1beta2/
-[6]: https://developers.google.com/datastore/docs/apis/v1beta2/proto
+[2]: java/demos/src/main/java/com/google/datastore/v1beta3/demos/trivial/Adams.java
+[4]: https://cloud.google.com/datastore
+[5]: https://cloud.google.com/datastore/reference/rest/index
```
Comment (pcostell) [inline README.md]: drop "index"

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -1,65 +1,130 @@
...
-v1beta2-rev1-3.0.4
-------------------
-- Make the dependency requirements in the Python client more permissive.
+  - Release notes for the API and `gcd` tool can be found at:
+    <https://cloud.google.com.com/datastore/docs/release-notes>
+  - Release notes are grouped by library.
+
+## Java Client
+
+### java-v1beta3-1.0.0
```
Comment (pcostell) [inline RELEASE_NOTES.md]: Create these tags --
java-v1beta3-1.0.0-beta
python-5.0.0-beta

File: RELEASE_NOTES.md [hunk 4]
```diff
@@ -1,65 +1,130 @@
...
+      - `datastore-v1beta3-protos`
+  - The versionIds no longer include the API version (e.g. `v1beta3`)
+    since it is now part of the artifactId.
+  - Updated `LocalDevelopmentDatastore` so it is in sync with the `gcd` tool.
+    - Fixes:
+      <https://github.com/GoogleCloudPlatform/google-cloud-datastore/issues/53>
+
+## Python Client
+
+### python-4.0.0b1 (v1beta3)
```
Comment (pcostell) [inline RELEASE_NOTES.md]: python-5.0.0-beta

File: RELEASE_NOTES.md [hunk 5]
```diff
@@ -1,65 +1,130 @@
...
+
+  - Support for [Google Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials).
+  - Fixes:
+    <https://github.com/GoogleCloudPlatform/google-cloud-datastore/issues/37>
+  - Environment variable `DATASTORE_DATASET_ID` is now `DATASTORE_PROJECT_ID`.
+  - Environment variable `DATASTORE_HOST` is now `DATASTORE_LOCAL_HOST` and
```
Comment (pcostell) [inline RELEASE_NOTES.md]: DATASTORE_EMULATOR_HOST

File: java/datastore/pom.xml [hunk 6]
```diff
@@ -1,78 +1,103 @@
...
+-->
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
-  <groupId>com.google.apis</groupId>
-  <artifactId>google-api-services-datastore-protobuf</artifactId>
+
+  <groupId>com.google.cloud.datastore</groupId>
+  <artifactId>datastore-v1beta3-proto-client</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
```
Comment (pcostell) [inline java/datastore/pom.xml]: 1.0.0-beta

File: java/datastore/pom.xml [hunk 7]
```diff
@@ -1,78 +1,103 @@
...
+  <name>${project.groupId}:${project.artifactId}</name>
+
   <packaging>jar</packaging>
-  <name>google-api-services-datastore-protobuf</name>
-  <url>https://github.com/GoogleCloudPlatform/google-cloud-datastore</url>
+
+  <parent>
+    <groupId>com.google.cloud.datastore</groupId>
+    <artifactId>datastore-v1beta3-proto-client-parent</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
```
Comment (pcostell) [inline java/datastore/pom.xml]: 1.0.0-beta

File: java/demos/pom.xml [hunk 8]
```diff
@@ -1,37 +1,70 @@
...
+  </parent>
+
   <dependencies>
     <dependency>
-      <groupId>com.google.apis</groupId>
-      <artifactId>google-api-services-datastore-protobuf</artifactId>
-      <version>v1beta2-rev1-4.0.0</version>
+      <groupId>com.google.cloud.datastore</groupId>
+      <artifactId>datastore-v1beta3-proto-client</artifactId>
+      <version>0.0.1-SNAPSHOT</version>
```
Comment (pcostell) [inline java/demos/pom.xml]: 1.0.0-beta

File: python/googledatastore/__init__.py [hunk 9]
```diff
@@ -14,13 +14,42 @@
...
+from google.datastore.v1beta3.query_pb2 import *
+
+from google.protobuf.timestamp_pb2 import Timestamp
+from google.protobuf.struct_pb2 import NULL_VALUE
+from google.rpc.status_pb2 import Status
+from google.rpc import code_pb2
+from google.type.latlng_pb2 import LatLng
+
+__version__ = '4.0.0b1'
+VERSION = (4, 0, 0, 'b1')
```
Comment (pcostell) [inline python/googledatastore/__init__.py]: 5, 0, 0

**Comment by pcostell** [comment]
Updated version numbers and tagged with appropriate version numbers.

**Comment by eddavisson** [comment]
LGTM

**Pull request merged**